src/rebar_prv_xref.erl

```diff
@@ -0,0 +1,289 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil -*-
+%% ex: ts=4 sw=4 et
+
+-module(rebar_prv_xref).
+
+-behaviour(provider).
+
+-export([init/1,
+         do/1,
+         format_error/1]).
+
+-include("rebar.hrl").
+
+-define(PROVIDER, xref).
+-define(DEPS, [compile]).
+
+%% ===================================================================
+%% Public API
+%% ===================================================================
+
+-spec init(rebar_state:t()) -> {ok, rebar_state:t()}.
+init(State) ->
+    Provider = providers:create([{name, ?PROVIDER},
+                                 {module, ?MODULE},
+                                 {deps, ?DEPS},
+                                 {bare, false},
+                                 {example, "rebar xref"},
+                                 {short_desc, short_desc()},
+                                 {desc, desc()}]),
+    State1 = rebar_state:add_provider(State, Provider),
+    {ok, State1}.
+
+-spec do(rebar_state:t()) -> {ok, rebar_state:t()} | {error, string()}.
+do(State) ->
+    ?INFO("Running cross reference analysis...", []),
+
+    %% Spin up xref
+    {ok, _} = xref:start(xref),
+    ok = xref:set_library_path(xref, code_path(State)),
+
+    xref:set_default(xref, [{warnings,
+                             rebar_state:get(State, xref_warnings, false)},
+                            {verbose, rebar_log:is_verbose(State)}]),
+
+    {ok, _} = xref:add_directory(xref, "ebin"),
+
+    %% Save the code path prior to doing anything
+    OrigPath = code:get_path(),
+    true = code:add_path(rebar_dir:ebin_dir()),
+
```

Comment:
Can you split up this function in some form of `init` function or something that could be sent out of the main flow? I.e. differentiate between spinning up and configuring xref and actually running it.